src/ansys/aedt/core/extensions/templates/run_pyaedt_script.py_build

```diff
@@ -41,60 +41,159 @@ sys.path.append(r"##EXTENSION_TEMPLATES##")
-            pyaedt_script_flag = pyaedt_utils.check_file(pyaedt_script, oDesktop)
-            if not pyaedt_script_flag:
-                return
-            # Add environment variables
-            pyaedt_utils.environment_variables(oDesktop)
-            # Run script
-            if is_linux:
-                pyaedt_utils.set_ansys_em_environment(oDesktop)
-                command = [
-                    python_exe,
-                    pyaedt_script,
-                    str(oDesktop.GetProcessID()),
-                    str(oDesktop.GetVersion()[:6]),
-                ]
-                my_env = os.environ.copy()
-                subprocess.Popen(command, env=my_env)
-            else:
-                command = [
-                    '"{}"'.format(python_exe),
-                    '"{}"'.format(pyaedt_script),
-                    str(oDesktop.GetProcessID()),
-                    str(oDesktop.GetVersion()[:6]),
-                ]
-                my_env = os.environ.copy()
-                subprocess.Popen(" ".join(command), env=my_env)
+        # --- GUI Selection ---
+        pyaedt_script, script_args = show_launcher_dialog(project_path)
+
+        if not pyaedt_script:
+            return
+
+        # Get AEDT version
+        version_short = oDesktop.GetVersion()[2:6].replace(".", "")
+        # CPython interpreter
+        python_exe = r"##PYTHON_EXE##" % version
+        # Check if CPython interpreter and AEDT release match
+        python_exe = pyaedt_utils.sanitize_interpreter_path(python_exe, version_short)
+
+        # Check Python executable
+        python_exe_flag = pyaedt_utils.check_file(python_exe, oDesktop)
+        if not python_exe_flag:
+            return
+        # Check script file
+        pyaedt_script_flag = pyaedt_utils.check_file(pyaedt_script, oDesktop)
+        if not pyaedt_script_flag:
+            return
+
+        # Add environment variables
+        pyaedt_utils.environment_variables(oDesktop)
+
+        # --- Run Script with Arguments ---
+        my_env = os.environ.copy()
+        python_exe = python_exe if is_linux else '"{}"'.format(python_exe)
+        pyaedt_script = pyaedt_script if is_linux else '"{}"'.format(pyaedt_script)
+        command = [
+            python_exe,
+            pyaedt_script,
+            script_args,  # Pass args
+        ]
+        command = command if is_linux else ' '.join(command)
+
+        oDesktop.AddMessage("", "", 0, command)
+
+        proc = subprocess.Popen(
+            command,
+            env=my_env,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+
+        stdout, stderr = proc.communicate()
+        for row in  stdout.splitlines():
+            oDesktop.AddMessage("", "", 0, row)
+        if proc.returncode != 0:
+            for row in stderr.splitlines():
+                oDesktop.AddMessage("", "", 2, row)
 
     except Exception as e:
         pyaedt_utils.show_error(str(e), oDesktop)
```

Comment:
same as above

Reply:
@hui-zhou-a 

Sorry, I think you are confused, this refactoring is to change the env. variables and remove the extra arguments passed in the Ironpython scripts.

But when we run the Run Script, why do we need quit() ? This script is closed once the PyAEDT script ends.